subworkflows/local/input_check.nf

```diff
@@ -5,22 +5,54 @@
 include { COMBINE_DATA } from '../../modules/local/combine_data.nf'
 include { fromSamplesheet } from 'plugin/nf-validation'
 
+
 workflow INPUT_CHECK {
 
     main:
 
-    // TODO add in automatic gzipping of all samples in
     versions = Channel.empty()
     def sample_sheet = params.input
+
+    // Thank you snvphylnfc for the ideas :)
+    // https://github.com/phac-nml/snvphylnfc/blob/f1e5fae76af276acf0a8c98174978cb21ca5d7e0/workflows/snvphylnfc.nf#L98-L109
+    def processedIDs = [] as Set
+
     reads_in = Channel.fromSamplesheet(
         "input", // apparentely input maps to params.input...
         parameters_schema: 'nextflow_schema.json',
         skip_duplicate_check: true).map {
             // Create grouping value
-            meta -> tuple(meta.id[0], meta[0])
+            meta ->
+
+                // Remove any unallowed charactars in the meta.id field
+                meta[0].id = meta[0].id.replaceAll(/[^A-Za-z0-9_\-]/, '_')
```

Comment:
You can remove this. `meta.id` only needs to be scrubbed of unallowed characters if `sample_name` is provided in the samplesheet. This relates to my next comment.